fortran/test/fetch_content_integration/test_micm_fort_api.F90

```diff
@@ -1,28 +1,42 @@
 program test_micm_fort_api
   use iso_c_binding
-  use micm_core, only: micm_t
+  use micm_core, only: micm_t, mapping_t
 
   implicit none
 
   type(micm_t), pointer         :: micm
   real(c_double)                :: time_step
   real(c_double)                :: temperature
   real(c_double)                :: pressure
-  integer(c_int)                :: num_concentrations
-  real(c_double), dimension(5)  :: concentrations
-  integer                       :: errcode
-  character(len=7)              :: config_path
+  integer(c_int)                :: num_concentrations, num_user_defined_reaction_rates
+  real(c_double), dimension(5)  :: concentrations 
+  real(c_double), dimension(3)  :: user_defined_reaction_rates 
+  integer                       :: errcode, i
+  character(len=256)            :: config_path
+  type(mapping_t)                 :: the_mapping
 
   time_step = 200
   temperature = 272.5
-  pressure = 101253.3
+  pressure = 101253.4
   num_concentrations = 5
   concentrations = (/ 0.75, 0.4, 0.8, 0.01, 0.02 /)
-  config_path = "chapman"
+  config_path = "configs/chapman"
+  num_user_defined_reaction_rates = 3
+  user_defined_reaction_rates = (/ 0.1, 0.2, 0.3 /)
+
 
   write(*,*) "[test micm fort api] Creating MICM solver..."
   micm => micm_t(config_path, errcode)
 
+  do i = 1, micm%species_ordering_length
+    the_mapping = micm%species_ordering(i)
+    print *, "Species Name:", the_mapping%name(:the_mapping%string_length), ", Index:", the_mapping%index
+  end do
+  do i = 1, micm%user_defined_reaction_rates_length
+    the_mapping = micm%user_defined_reaction_rates(i)
+    print *, "User Defined Reaction Rate Name:", the_mapping%name(:the_mapping%string_length), ", Index:", the_mapping%index
+  end do
+
```

Comment:
I'm curious how we can do this for the CCPP wrapper. Would CAM-SIMA take care of this?

Reply:
We will have to use these in cam sima to properly order things